.github/workflows/deploy-pull-request.yml

```diff
@@ -0,0 +1,47 @@
+name: Deploy PR to octoboard-development
+
+on:
+  pull_request:
+    types: [labeled, opened, synchronize, reopened]
+
+jobs:
+  deploy:
+    name: Deploy server
+    runs-on: ubuntu-latest
+    if: contains(github.event.pull_request.labels.*.id, 2818529465)
```

Comment:
This will only deploy if PR has `preview-deploy` label. Another option would be to run deploys on demand (for example triggered by a comment on PR)

Reply:
I think the solution with label is very good 👍